blog.qmd

```diff
@@ -0,0 +1,329 @@
+we explored how we gathered course descriptions from community colleges in
+Florida using web scraping techniques. Work-based learning (WBL), which consists
+of opportunities such as internships, apprenticeships, and practicums, focuses
+on career preparation and training in supervised, work-like settings. Community
+colleges, with their career-focused curricula and diverse student populations,
+offer an ideal setting for such opportunities. We successfully compiled a
+comprehensive dataset of course descriptions, but the question remained: what is
+the prevalence and nature of WBL opportunities in Florida's
+community colleges as reflected in their course descriptions?
+
+To answer this question, we turned to text analysis, a methodology that allows us
+to extract meaningful information from large volumes of text data. Text analysis
+is particularly suited to our research question because it enables us to
+identify and quantify specific keywords, such as terms related to work-based
+learning, within our data frame of course descriptions. In this post, we'll
+delve into our methodology for analyzing this data using the `quanteda` package
+in R, a powerful tool for quantitative text analysis.
+
+## Getting Started with Quanteda
+
+[Quanteda](https://quanteda.io), short for Quantitative Analysis of Textual
+Data, is a powerful tool for managing and analyzing text data in R. It offers a
+suite of functions for managing a
+[corpus](https://tutorials.quanteda.io/basic-operations/corpus/corpus/)---or a
+collection of texts---such as creating [document-feature
+matrices](https://tutorials.quanteda.io/basic-operations/dfm/), [analyzing
+keywords](https://tutorials.quanteda.io/basic-operations/tokens/kwic/), and
+more. These functions are highly efficient and provide a consistent interface
+with support for multiple languages. While it operates as a standalone package,
+it also integrates seamlessly with extensions such as `readtext`, `spacyr`, and
+`quanteda.textstats`.
+
+To install and load the required packages, we can use the librarian package for
+convenience
+
+```{r text-}
+librarian::shelf(tidyverse, # data wrangling
+                 quanteda, # text mining
+                 readtext # read texts and associated document-level meta-data
+)
+```
+
+Next, we load our text data containing course descriptions as well as
+document-level metadata using the `readtext::readtext()` function. We then
+perform some additional cleaning steps to standardize variable names and focus
+our analysis on active courses for credit. Here's a brief explanation of each
+column in our courses dataframe:
+
+-   doc_id: This is a unique identifier for each course. It is constructed by
+    combining the school abbreviation, the course prefix, and the course number.
+    For example, 'BC-THE-2300' refers to a course at the school abbreviated as
+    'BC', with the course prefix 'THE', and the course number '2300'.
+
```

Comment:
I would just name the school and course here to be more concrete! But this is super helpful

Reply:
Can you actually say the full name of the school and course here? I think the fully spelled out example makes this more accessible. I.e. "refers to [course name] 2300 at [school name]"

Reply:
Yeah that format makes more sense to me too